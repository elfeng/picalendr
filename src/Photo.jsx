import React from 'react';
import moment from 'moment';
import './Photo.css';

export function formatDateTaken(dateTakenAsSeconds) {
    const dateTaken = moment(dateTakenAsSeconds * 1000);
    return dateTaken.format('LL');
}

export default function Photo(props) {
    return (
        <div className="photo">
            <a href={props.link}>
                <img src={props.url} alt={props.search}/>
                <div>
                    Taken on: {formatDateTaken(props.dateTaken) }
                </div>
            </a>
        </div>
    );

};