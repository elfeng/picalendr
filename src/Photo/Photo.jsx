import React from 'react';
import './Photo.css';

export default function Photo(props) {
    return (
        <div className="photo">
            <a href={props.linkUrl}>
                <img src={props.snapshotUrl} alt={props.title}/>
                <div>
                    Taken on: {props.dateTakenFormatted}
                </div>
            </a>
        </div>
    );

};