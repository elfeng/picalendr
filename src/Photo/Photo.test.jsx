import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
import renderer from 'react-test-renderer';
import Photo from './Photo';

it('renders without crashing', () => {
  const testUtilsRendered = ReactTestUtils.createRenderer();
  testUtilsRendered.render(<Photo />);
  const result = testUtilsRendered.getRenderOutput();
  expect(result.type).toBe('div');
});

it('should match the snapshot', () => {
  const component = renderer.create(
    <Photo
      linkUrl="https://www.flickr.com/photos/21668212@N02/25277702294"
      snapshotUrl="https://farm2.staticflickr.com/1658/25277702294_3514fc66de.jpg"
      title="alpagas at sajama"
      dateTakenFormatted="April 1, 2015"/>
      );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

