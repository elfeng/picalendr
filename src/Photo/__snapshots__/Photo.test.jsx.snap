exports[`test should match the snapshot 1`] = `
<div
  className="photo">
  <a
    href="https://www.flickr.com/photos/21668212@N02/25277702294">
    <img
      alt="alpagas at sajama"
      src="https://farm2.staticflickr.com/1658/25277702294_3514fc66de.jpg" />
    <div>
      Taken on: 
      April 1, 2015
    </div>
  </a>
</div>
`;
